torch/serialization.py

```diff
@@ -742,6 +746,15 @@ def load(
         # Load a module with 'ascii' encoding for unpickling
         >>> torch.load('module.pt', encoding='ascii')
     """
+    UNSAFE_MESSAGE = (
+        "*WARNING* Safe load failed, defaulting to standard Python pickle module, "
```

Comment:
It's seems weird to have a mode which tries safe load, and then AUTOMATICALLY tries the unsafe load after. Like, what's the point? It's not any more secure, and it also gives you a false sense of security when there is no warning because the safe load succeeded.

Reply:
Also thought it was strange to go with the unsafe load if safe failed, I suppose it's to avoid causing breaking changes for backwards compat.

Reply:
I guess the question is, do we care about backward compatibility or not. I'm of two minds to be frank, but fine with flipping the logic to `load` should  be called with `weights_only` set to `False`

Reply:
We do care about BC... but if you want to deprecate the old insecure API, you should raise the warning unconditionally.

Reply:
> It's seems weird to have a mode which tries safe load, and then AUTOMATICALLY tries the unsafe load after.

@ezyang - note that this warning as written currently in the diff gets triggered only if `weights_only=False`. So it doesn't violate safety.

This behavior (as implemented) makes sense only as a temporary "one release" workaround to drive people to put explicit `weights_only=False` in the needed places. I.e. those would be places where this logic fails and generates a warning. Then in the follow up release we'd switch default to be `weights_only=True`. Places that were handling tensors only anyway (presumably the majority) won't need to change the code, will never see the warning and will become secure in the future.

Of course we can jump to `weights_only=True` right away but I'm a bit hesitant about breaking BC